lego/apps/companies/views.py

```diff
@@ -48,11 +48,17 @@ class AdminCompanyViewSet(AllowedPermissionsMixin, viewsets.ModelViewSet):
     queryset = (
         Company.objects.all()
         .prefetch_related("semester_statuses", "files")
-        .select_related("student_contact")
     )
     pagination_class = None
     permission_handler = CompanyAdminPermissionHandler()
 
+    def _list(self, request, *args, **kwargs):
+        semester_id = request.query_params.get('semester_id', None)
+
+        serializer = self.get_serializer(self.queryset, many=True, context={'semester_id': semester_id})
```

Comment:
Is there some way to just update the context (so we don't lose request and the other couple things passed down by default)?

Reply:
Good point! I'll look into it